tools/build-toolchain.sh

```diff
@@ -87,30 +143,34 @@ cd gcc_compile
   --disable-win32-registry \
   --disable-nls \
   --disable-werror \
-  --with-system-zlib
+  --with-system-zlib \
+  $CROSS_COMPILE_FLAGS
 make all-gcc -j "$JOBS"
 make all-target-libgcc -j "$JOBS"
 make install-gcc || sudo make install-gcc || su -c "make install-gcc"
 make install-target-libgcc || sudo make install-target-libgcc || su -c "make install-target-libgcc"
+echo "Finished Compiling GCC-$GCC_V for MIPS N64 (pass 1) outside of the source tree"
```

Comment:
I think the idea of having 2 GCC passes was to support exactly this scenario. So let's try something different.

The first two steps (binutils+GCC) should build a compiler for the **build** platform, so for Linux. Don't add your `$CROSS_COMPILE_FLAGS` here. This is what will build and install the linux->mips compiler that you need later to avoid the error. Then you build newlib. That will be done with the linux->mips compiler, so specifying `$CROSS_COMPILE_FLAGS` should not be required as well.

Now, before the final pass (GCC step 2), you probably need to add a new pass to build binutils *again*, but this time with $CROSS_COMPILE_FLAGS. This will build the windows->mips binutils which are required. And then, the final pass with GCC with $CROSS_COMPILE_FLAGS, that should succeed because the step 1 has installed the pre-required linux->mips compiler.

Hope this makes sense.

To try this, please also modify the CI to have a separate job that *only* does the Windows build.

Reply:
So to better clarify. Assume that you have defined two variables: `BUILD=x86_64-linux-gnu` and `HOST` that can be either `x86_64-linux-gnu` or `x86_64-w64-mingw32`. 

I think the workflow should be (pseudo-code)

```
if BUILD != HOST {
     Build binutils BUILD->TARGET
     Build GCC BUILD->TARGET (no C++, only all-gcc/all-libgcc)  -- step1
}

Build binutils BUILD/HOST/TARGET
Build GCC BUILD/HOST/TARGET with C,C++  -- step 2
Build newlib for TARGET
```

So this is actually exactly what I wrote in the previous comment, but it's just that checks are differents. You want an initial phase which is run only when BUILD != HOST which builds binutils+GCC as basis, and then the main phase which is always run that builds binutils+GCC+newlib for the BUILD/HOST/TARGET triplet.

If you structure like this, you can probably drop the custom command line and simply having the CI redefine the HOST environment. Something like:

```
BUILD=x86_64-linux-gnu
HOST=${HOST:-$BUILD}
TARGET=mips64-elf
```

and then the CI should be able to do `HOST=x86_64-w64-mingw build-toolchain.sh`